SalesforceCommerceCloud/pwa-kit
# @W-14446652@ Change CSP enforcement to an opt-in middleware.

In v3.2.0, to facilitate adoption of Storefront Preview, we introduced logic to enforce a Content-Security-Policy header containing all the directives that are required for PWA Kit / Storefront Preview to function. However, while making CSP management easier for those who use it, this is a breaking change for users who explicitly do _not_ use the Content-Security-Policy header. 🤕 Oops.

This PR simply removes the middleware function from the internal logic of `pwa-kit-runtime` and exports it so that end users must opt in to using the function.

# Description



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

- Check out v3.2.0, launch the app, save the CSP header.
- Check out this code, launch the app. CSP header should be the same.
- Remove the two security header middleware functions. Restart the app. CSP header should not be present.
https://github.com/SalesforceCommerceCloud/pwa-kit/blob/3582944d8fc0053ac19022e0637610646f494384/packages/template-retail-react-app/app/ssr.js#L38-L60
- Bonus fun: move the `defaultSecurityHeaders` middleware to after the `helmet` middleware and see that it doesn't change the CSP!

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
